init/cli/lib.go

```diff
@@ -146,16 +151,37 @@ func getConfiguredCommands(ctx cli.Context, loggers launchlib.ServiceLoggers) (m
 	return cmds, nil
 }
 
-func isPidRunning(pid int) (bool, *os.Process) {
+func isPidRunning(pid int) (bool, *os.Process, error) {
 	// Docs say FindProcess always succeeds on Unix.
 	proc, _ := os.FindProcess(pid)
-	if isProcRunning(proc) {
-		return true, proc
+	running, err := isProcRunning(proc)
+	if err != nil {
+		return false, nil, err
 	}
-	return false, nil
+	if running {
+		return true, proc, nil
+	}
+	return false, nil, nil
 }
 
-func isProcRunning(proc *os.Process) bool {
+func isProcRunning(proc *os.Process) (bool, error) {
 	// This is the way to check if a process exists: https://linux.die.net/man/2/kill.
-	return proc.Signal(syscall.Signal(0)) == nil
+	// On linux, this may respond true if there is a thread running with the same id.
+	running := proc.Signal(syscall.Signal(0)) == nil
+	if !running {
+		return false, nil
+	}
+
+	// On linux, iterating over processes will only return running processes. Unfortunately,
+	// getting exactly the one pid will also return a thread of the same id.
```

Comment:
> Unfortunately, getting exactly the one pid will also return a thread of the same id.

Is there actually a difference between calling `ps.Processes()` then manually looking for a pid, vs calling `findProcess(pid)`?

From the [process_unix.go](https://github.com/mitchellh/go-ps/blob/master/process_unix.go#L36-L90) of this library it seems to me that both will end up doing the same thing, which is looking in `/proc/` for a directory(ies) matching a process number, and then wrap that with `newUnixProcess`.

Reply:
Yes. I tested it manually, they are different.

From man proc:
```
The /proc/[tid] subdirectories are not visible when iterating through /proc with getdents(2) (and thus are not visible when one uses ls(1) to view the contents of /proc).
```